nemo/core/connectors/save_restore_connector.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle
+import shutil
+import tarfile
+import tempfile
+import uuid
+from os import path
+from typing import Optional, Union
+
+import torch
+from omegaconf import DictConfig, OmegaConf
+from omegaconf.omegaconf import open_dict
+
+from nemo.utils import app_state, logging, model_utils
+from nemo.utils.app_state import AppState
+from nemo.utils.model_utils import import_class_by_path
+
+
+class SaveRestoreConnector:
```

Comment:
I feel this class has so many methods starting with `_`. If they should be used outside the class, they should not have `_` prefix

Reply:
Done.